### PR TITLE
apps page

### PR DESCRIPTION
**Pull request by will-moore**
Added new apps page at  /omero/apps/

Staged at https://snoopycrimecop.github.io/www.openmicroscopy.org/omero/apps/

cc @jburel

### CODE REVIEW

File: omero/apps/index.html [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+            </div>
+        </div>
+        <div class="row">
+            <div class="medium-1 medium-offset-2 columns">
+                <i class="icon-fa-blue fa fa-th fa-2x"></i>
+            </div>
+            <div class="omero-key-feature-container medium-7 columns end">
+                <h4>OMERO-metadata</h4>
+                <p>
+                    Covert tabular data from CSV files to OMERO.tables.
```
Comment (mtbc) [inline omero/apps/index.html]: Convert

File: omero/apps/index.html [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+
+        <hr class="whitespace">
+
+        <div class="row column text-center">
+            <h2>Command-line plugins</h2>
+        </div>
+        <div class="row column text-center">
+            <p>
+                The OMERO command-line is a powerful tool that can be extended with
```
Comment (mtbc) [inline omero/apps/index.html]: no hyphen for this one

**Comment by joshmoore** [comment]
A few high-level comments:

 * What pages will link to this page?
   - Is it intended to have navigation a la "Institutions" and "Features"?
   - Or perhaps "Extend" under https://snoopycrimecop.github.io/www.openmicroscopy.org/omero/?  
   - As well as under "omero/developers".
   - Under "omero/scientists" and "omero/institutions" it could says "Apps".
   - (Not sure what the logo would be.)
   - Also perhaps "docs/" could someone say, "see each separate app for its own docs".
   - And finally to get really fancy, we could embed apps in "explore/"
 * Was a grid layout attempted considering that's what is used elsewhere? (thrown out because of the ultimate length? or perhaps not tried since getting content first?)
 * Possible additional content:
  - links back to the general documentation on how to install (if available)
  - for web apps, an indication of which apps can be embedded in other pages

**Comment by will-moore** [comment]
@joshmoore 
I was going to add some links to this page, but then I had no idea what we want, so I punted (I meant to ask that in description). 
I think it needs to be quite visible from https://snoopycrimecop.github.io/www.openmicroscopy.org/omero/ but that really needs a bit of design thought.

I went for a list rather than a grid because it felt right for a "listing" of apps, kinda like https://pypi.org/search/?q=omero
Also it allows us to add more or less text to each one without worrying about grid layout issues as in https://github.com/openmicroscopy/www.openmicroscopy.org/pull/245

Not sure what you mean "links back to the general documentation on how to install" since they mostly link to the pypi pages. I guess figure and iviewer could link to pypi AND to their home pages?

For web apps, I'm not sure any can be embedded in other pages (except in iframe) except maybe webgateway viewer. Or do you mean embed in webclient (like parade)?

**Comment by joshmoore** [comment]
> links back to the general documentation on how to install

Sorry, I meant more our generalized "server/web plugins" documentation, assuming there is such a central page with info on the venv, etc.

> Or do you mean embed in webclient (like parade)?

No, I meant like webgateway viewer. Ok. I didn't realize none of the new apps had public JS API for that type of thing.

**Comment by will-moore** [comment]
I wonder if we want to consistently name 'cli' plugins differently from others? I guess we haven't done it with web.
e.g. uses might not look for ```OMERO-cli-render``` alphabetically before ```OMERO-metadata```?

**Comment by will-moore** [comment]
I'm not sure our general documentation mentions web apps anywhere, e.g. http://docs.openmicroscopy.org/omero/5.4.7/sysadmins/unix/install-web/walkthrough/omeroweb-install-centos7-ice3.6.html
We've always assumed that sysadmins know where to run ```pip install ...```.

**Comment by joshmoore** [comment]
* http://docs.openmicroscopy.org/omero/5.4.7/developers/Web/WebclientPlugin.html?highlight=apps
 * http://docs.openmicroscopy.org/omero/5.4.7/developers/Web/CreateApp.html?highlight=apps
 * http://docs.openmicroscopy.org/omero/5.4.7/developers/cli/extending.html?highlight=extensions

**Comment by pwalczysko** [comment]
I would think OMERO.IVIEWER -> OMERO.iviewer, OMERO.FIGURE -> OMERO.figure etc.

**Comment by pwalczysko** [comment]
Bug

The link next to the cli-render plugin in https://snoopycrimecop.github.io/www.openmicroscopy.org/omero/apps/ brings me to OMERO.iviewer

**Comment by joshmoore** [comment]
> I would think OMERO.IVIEWER -> OMERO.iviewer, OMERO.FIGURE -> OMERO.figure etc.

https://snoopycrimecop.github.io/www.openmicroscopy.org/omero/features/organize/ have webtagging & forms large, so this may be a larger issue.

Possibly related: https://github.com/openmicroscopy/www.openmicroscopy.org/pull/106

**Comment by will-moore** [comment]
I fixed the App names and links to lowercase in /omero/apps/ page.
Also added a link to this page from the /omero/ page (last item in the grid). This might not be prominent enough. Any other suggestions for where to add links?

**Comment by pwalczysko** [comment]
All good (links and capitalization) except

>Also added a link to this page from the /omero/ page (last item in the grid). 

This seems simply not to have been pushed.

**Comment by will-moore** [comment]
@pwalczysko Apologies - pushed now.

**Comment by will-moore** [comment]
To test last commits:
 - https://snoopycrimecop.github.io/www.openmicroscopy.org/omero/ should have a link to apps (last item in grid)
 - https://snoopycrimecop.github.io/www.openmicroscopy.org/omero/developers/ should have link to apps (under OMERO.web item)
 - https://snoopycrimecop.github.io/www.openmicroscopy.org/omero/apps/ now lists apps alphabetically.

**Comment by mtbc** [comment]
Would it make sense to have a little bit more horizontal space between icons and descriptions?
app

**Comment by will-moore** [comment]
This issues also affects other pages, particularly when the screen is small (but not 'mobile' layout):
 - /about/
 - /commercial-partners/
 - /news/
 - /explore/

Should be fixed by a83a4d3.
To test, compare apps pages and those others in this PR vv main website.

**Comment by will-moore** [comment]
Also, added links for Apps page and Training page to the site-map: https://snoopycrimecop.github.io/www.openmicroscopy.org/site-map/

**Comment by joshmoore** [comment]
> Also, added links for Apps page and Training page to the site-map: 

:+1: but is `Add Ons` now a dupe?

**Comment by will-moore** [comment]
@joshmoore Not really, since all pages should be linked to from site-map, even if links to those pages exist elsewhere.

**Comment by joshmoore** [comment]
Except `Add-ons` doesn't exist, no?

**Comment by will-moore** [comment]
There's no ```Add-ons``` section of the site. It's just a header on the apps page to set the iviewer and figure links apart from other OMERO links, and I think this is still needed. We haven't added an "Apps" section to the site, just an apps page. I guess we could replace the ```Add-ons``` header with "Apps" and have that link to the omero/apps/ page? I'll try that...

**Comment by will-moore** [comment]
To test latest changes:
 - check that the layout issue in https://github.com/openmicroscopy/www.openmicroscopy.org/pull/249#issuecomment-423926574 is fixed in apps page and others listed in https://github.com/openmicroscopy/www.openmicroscopy.org/pull/249#issuecomment-424650785
 - Check 'Apps' and 'Training' links in site-map.

**Comment by joshmoore** [comment]
Sorry I don't see why some files are listed explicitly under Apps and all aren't. This might be something just needs a general review later.

File: omero/apps/index.html [hunk 3]
```diff
@@ -0,0 +1,163 @@
+---
+layout: default
+title: OMERO Training
```
Comment (sbesson) [inline omero/apps/index.html]: Why this title?

**Comment by will-moore** [comment]
We can't list more pages under Apps on the site-map until those pages get created. We don't (yet) have pages for gallery, parade, fpbioimage or mapr. There are pypi pages for those apps, but external pages shouldn't appear on a site-map.

File: site-map/index.html [hunk 4]
```diff
@@ -133,12 +134,13 @@ <h4 class="subdivider">Key Features</h4>
...
         <div class="row">
             <div class="small-12 medium-9 medium-offset-3 columns">
-                <h4 class="subdivider">Add Ons</h4>
+                <a href="{{ site.baseurl }}/omero/apps/"><h4 class="subdivider">Apps</h4></a>
             </div>
         </div>
         <div class="row">
             <div class="small-12 medium-3 medium-offset-3 columns">
                 <ul class="no-list-style">
+                    <li><a href="{{ site.baseurl }}/omero/apps/">Apps</a></li>
```
Comment (sbesson) [inline site-map/index.html]: In other OMERO subsections, the landing page is listed as `Summary`

**Comment by joshmoore** [comment]
> We can't list more pages under Apps on the site-map until those pages get created. We don't (yet) 

Sorry, I mean those 2 pages are linked from under Apps. Under Training for example, we don't list all the subpages. Either we need to review and add all subpages or we don't everywhere.

File: omero/apps/index.html (hunk 3 above)
Comment (will-moore) [reply omero/apps/index.html]: Because I copied and pasted from the training page to start the apps page and never renamed the title.

**Comment by will-moore** [comment]
@sbesson Sorry, I'm not sure what changes you're suggesting with "Given the site map, it feels exposing the apps alongside features & developers would be an natural option".
Do you want to add APPS to the top links OVERVIEW DOWNLOADS SCIENTISTS DEVELOPERS INSTITUTIONS FEATURES of each OMERO page?

**Comment by will-moore** [comment]
@joshmoore Added a link to the other Training page to the site-map.

**Comment by sbesson** [comment]
@will-moore I think this partly comes back to the questions raised in https://github.com/openmicroscopy/www.openmicroscopy.org/pull/249#issuecomment-414988674. This is certainly my expectation from looking at the structure under https://snoopycrimecop.github.io/www.openmicroscopy.org/site-map/.

I don't have enough context about the roadmap for promoting the Apps on the website, it might be we want the page to exist first and explore how to integrate into the navigation later.

**Comment by will-moore** [comment]
I also have a suggestion for linking to the apps page from web install docs: https://github.com/openmicroscopy/ome-documentation/pull/1906

**Comment by will-moore** [comment]
@jburel Any thoughts about links to the apps page? Its always tempting when adding a new page to make it prominent and add lots of links, but if you do this for all pages its not so sustainable.

**Comment by jburel** [comment]
minimally: It should be linked to:

* https://snoopycrimecop.github.io/www.openmicroscopy.org/omero/ (it is)
* https://snoopycrimecop.github.io/www.openmicroscopy.org/site-map/ (it is)
* The page could be available under ``products/omero apps``

The main problem it that we currently did not have a page with a "recap" of the available extensions. We have info but it is scattered around.
So Users need to have easy access to it

I think we will need to explore a bit more how to integrate it in the navigation e.g. added to dev/institutions page.

**Comment by will-moore** [comment]
The /apps page is also linked from https://snoopycrimecop.github.io/www.openmicroscopy.org/omero/developers/

**Pull request merged**